ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,363 @@
+            associated_files (dict): needed to find metadata files (provided by plugin)
+            imageset_paths (list): filepaths to all imagesets being looked at (provided by 'create' input)
+        """
+        # Gets metadata prefix and suffix
+        metadata_prefix = self.associated_files['metadata'][0]
+        metadata_suffix = self.associated_files['metadata'][1]
+        if metadata_suffix != '.json':
+            raise Exception("Currently non-json metadata files are not supported for the default loading of image ids")
+        
+        # Goes through each file in each imageset to search for metadata files
+        # metadata files are parsed for tags and filename is parsed for image_id 
+        for data_dir in self.imageset_paths:
+            for dir_entry in os.scandir(data_dir):
+                if not dir_entry.name.startswith(metadata_prefix):
+                    continue
+                image_id = dir_entry.name.replace(metadata_prefix, '').replace(metadata_suffix, '')
+                temp = read_json_metadata(dir_entry, image_id)
+                self.tags_df = pd.concat((self.tags_df, temp), sort=False)
+            self.tags_df = self.tags_df.fillna(False)
+
+        self.image_ids = self.tags_df.index.tolist()
+
+    def interactive_filter(self):
+        """Method is expected to only be called after 'load_image_ids' is called, as it relies on 
+            'self.tags_df' to be prepopulated. Method prompts user through interactive filtering 
+            of image_ids based on their tags.
+
+            If overloaded, method is expected to set 'self.image_ids' to whatever image_ids are still
+            to be used after filtering. 'self.filter_metadata' also needs to be set to a dict containing
+            set-names as keys and lists of image_ids as values.
+
+        Variables Needed:
+            tags_df (dataframe): needs to be set for the 'default_filter' function to be able to filter by tags
+                (provided by 'load_image_ids')
+        """
+        self.image_ids = default_filter(self.tags_df, self.filter_metadata)
+
+    def load_data(self):
+        """Method is expected to be called after 'load_image_ids' and 'interactive_filter' if filtering is
+            desired. Method goes through each image_id and copies its corresponing files into a temp directory
+            which will be later used by the plugin to create their dataset.
+
+            If overloaded, method is expected to copy all files the plugin needs into the provided 'temp_dir'.
+
+        Variables Needed:
+            image_ids (list): needed to find what needs to be copied (provided by 'load_image_ids'/'interactive_filter')
+            temp_dir (Path): needed to know where to copy to (provided by 'create' input)
+            associated_files (dict): needed to know what files need to be copied (provided by plugin)
+        """
+        copy_data_locally(self.image_ids, self.temp_dir, self.associated_files)
```

Comment:
@cSchubes @kevin3-black This is where data is copied into a temp directory. Pros of changing to sending plugins paths to the actual data is that it wastes less resources. Cons are that it is possible for plugins to screw up the data (although only if they add files to the imageset caches since `s3 sync` will take care of changed or deleted files). I don't feel too strongly on the topic, but if it does save a significant amount of time for larger datasets it may be worth it since it doesn't seem too difficult to implement.